go/adbc/driver/snowflake/driver.go

```diff
@@ -330,6 +330,7 @@ func (d *database) SetOptions(cnOptions map[string]string) error {
 
 	var err error
 	for k, v := range cnOptions {
+		v := v // copy into loop scope
```

Comment:
since the loop var here is a string and not a pointer, were we actually hitting this foot gun? Strings in Go are immutable so I don't *think* we were hitting this issue and thus this isn't necessary, but I could be wrong

Reply:
Okay, I guess we must have been hitting this or else this entire PR wouldn't have been needed lol, very odd